test/e2e/storage/volume_io.go

```diff
@@ -0,0 +1,822 @@
+	}
+
+	// create files of the passed-in file sizes
+	for _, fsize := range fsizes {
+		// file sizes must be a multiple of the length of the `writeBlk` value (1024 currently)
+		if fsize < len(writeBlk) {
+			fsize = len(writeBlk)
+		} else if fsize/len(writeBlk)*len(writeBlk) != fsize {
+			fsize = fsize/len(writeBlk) + len(writeBlk)
+		}
+		fpath := path.Join(dir, fmt.Sprintf("%s-%d", file, fsize))
+
+		By(fmt.Sprintf("Writing %d bytes to test file %s", fsize, fpath))
+		// dd's bs*seek = size (note this creates the file 1 byte more than the target size)
+		seek := fsize/len(writeBlk) - 1
+		writeCmd := fmt.Sprintf("dd if=%s of=%s bs=%d seek=%d", dd_input, fpath, len(writeBlk), seek)
+		_, err = podExec(clientPod, writeCmd)
+		if err != nil {
+			return fmt.Errorf("failed to write %d bytes to file %s: %v", fsize, fpath, err)
+		}
+		defer func(delFile string) {
+			By(fmt.Sprintf("deleting test file %s...", delFile))
+			_, rmErr := podExec(clientPod, fmt.Sprintf("rm -f %s", delFile))
+			if rmErr != nil {
+				framework.Logf("Unable to delete test file %s: %v", delFile, rmErr)
+				if err == nil { // rm err is returned if err is not set
+					err = rmErr
+				}
+			}
+		}(fpath)
+
+		By("verifying file size")
+		rtnstr, err := podExec(clientPod, fmt.Sprintf("stat -c %%s %s", fpath))
+		if err != nil || rtnstr == "" {
+			return fmt.Errorf("unable to get file size via `stat %s`: %v", fpath, err)
+		}
+		size, err := strconv.Atoi(strings.TrimSuffix(rtnstr, "\n"))
+		if err != nil {
+			return fmt.Errorf("unable to convert string %q to int: %v", rtnstr, err)
+		}
+		if size-1 != fsize {
+			return fmt.Errorf("size of file %s is %d, expected %d", fpath, size-1, fsize)
+		}
+
+		By("verifying file content")
+		rtnstr, err = podExec(clientPod, fmt.Sprintf("grep -c -m1 %q %s", writeBlk, fpath))
+		if err != nil {
+			return fmt.Errorf("unable to test file content via `grep %s`: %v", fpath, err)
+		}
+		foundCnt, err := strconv.Atoi(strings.TrimSuffix(rtnstr, "\n"))
+		if err != nil {
+			return fmt.Errorf("unable to convert string %q to int: %v", rtnstr, err)
+		}
+		if foundCnt == 0 {
+			return fmt.Errorf("Expected string %q to be found in file %s", writeBlk, fpath)
+		}
+	}
+
+	return
+}
+
+// These tests need privileged containers which are disabled by default.
+var _ = framework.KubeDescribe("Volume plugin streaming [Volume][Slow]", func() {
+	f := framework.NewDefaultFramework("volume-io")
+	var (
+		config    framework.VolumeTestConfig
+		cs        clientset.Interface
+		ns        string
+		serverIP  string
+		serverPod *v1.Pod
+	)
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace.Name
+	})
+
+	////////////////////////////////////////////////////////////////////////
+	// NFS
+	////////////////////////////////////////////////////////////////////////
+	framework.KubeDescribe("NFS", func() {
+		fileSizes := []int{1024, 1048576, 1073741824} // 1kib, 1mib, 1gib
```

Comment:
I found base 10 const in _e2e_node/summary_test.go_. I moved those to the framework package and added base2 constants to _framework.go_.